compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -488,12 +488,32 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                                     // don't create labels for compiler-generated spans
                                     Some(_) => None,
                                     None => {
-                                        let (span, suggestion) = suggest_ampmut(
-                                            self.infcx.tcx,
-                                            local_decl,
-                                            opt_assignment_rhs_span,
-                                            *opt_ty_info,
-                                        );
+                                        let (span, suggestion) = if name != Symbol::intern("self") {
```

Comment:
Going to change this to
```suggestion
                                        let (span, suggestion) = if name != kw::SelfLower  {
```